src/services/ocrService.ts

```diff
@@ -93,8 +94,8 @@ export class OCRService {
                     ]
                 );
                 body = bodyAndType[0];
-                const fileType = bodyAndType[1].mime;
-                headers = { "Content-Type": fileType, "cache-control": "no-cache" };
+                const contentType = bodyAndType[1]?.mime ? bodyAndType[1].mime : mimeType;
```

Comment:
How is this change correlated with the major change of this PR? Is it necessary change? or just to make things better?

Reply:
yeah it's related. Without this, users cannot run ocr on local files because the ocr API requires correct mime-type/content-type when using files as source (not a url)

Reply:
So we can pass the mime-type/content-type already determined previously. Maybe it still needs some optimization though